src/components/table/DatasetTable.tsx

```diff
@@ -1,122 +1,114 @@
 import React from 'react';
-import { WithStyles, withStyles } from '@mui/styles';
 import moment from 'moment';
 import { DatasetSummaryModel } from 'generated/tdr';
 import { OrderDirectionOptions, TableColumnType } from 'reducers/query';
-import { CustomTheme } from '@mui/material/styles';
 
 import TextContent from 'components/common/TextContent';
+import { styled } from '@mui/system';
 import { renderCloudPlatforms } from '../../libs/render-utils';
 import LightTable from './LightTable';
 import ResourceName from './ResourceName';
 import { ResourceType } from '../../constants';
 
-const styles = (theme: CustomTheme) => ({
-  textWrapper: {
-    ...theme.mixins.ellipsis,
-  },
-});
+const TextWrapper = styled('span')(({ theme }) => ({
+  ...theme.mixins.ellipsis,
+}));
 
-interface IProps extends WithStyles<typeof styles> {
-  datasets: Array<DatasetSummaryModel>;
-  datasetRoleMaps: { [key: string]: Array<string> };
-  datasetsCount: number;
-  filteredDatasetsCount: number;
-  handleFilterDatasets?: (
+interface IProps {
+  readonly datasets: Array<DatasetSummaryModel>;
+  readonly datasetRoleMaps: { [key: string]: Array<string> };
+  readonly datasetsCount: number;
+  readonly filteredDatasetsCount: number;
+  readonly handleFilterDatasets?: (
```

Comment:
Apologies if I've asked this before, but why do we need readonly tag on the types?